metaflow/plugins/aip/aip_metaflow_step.py

```diff
@@ -435,14 +437,18 @@ def aip_metaflow_step(
         with open(output_file, "w") as f:
             f.write(str(values[idx]))
 
-    # get card and write to output file
-    _write_card_artifacts(
-        flow_name,
-        step_name,
-        task_id,
-        passed_in_split_indexes,
-        metaflow_run_id,
-    )
+    # Write card manifest (html) to Argo output artifact path.
+    try:
```

Comment:
can `add_default_cards` condition be checked here, and then we no longer need to catch the exception?

Reply:
This try catch is to ignore the rendering failures

Reply:
would it be better not to call `_write_card_artifacts` if `add_default_cards` is false, and if `add_default_cards` is true to not ignore the failure to `_write_card_artifacts`?

Reply:
I'll make sure the care rendering is not called if the step has no card decorator.

Reply:
This is not a problem as `_write_card_artifacts` is safe with empty list of cards.

Reply:
Current behavior:
- If no card is expected, skip card rendering.
- Otherwise logs all exceptions but catching expect raising. 
  - The write failure simply means cards are not added as Argo artifacts, but are still visible through Metaflow UI or CLI